src/components/FullscreenLoadingIndicator.js

```diff
@@ -5,17 +5,22 @@ import {ActivityIndicator, StyleSheet, View} from 'react-native';
 import styles from '../styles/styles';
 import themeColors from '../styles/themes/default';
 import stylePropTypes from '../styles/stylePropTypes';
+import Log from '../libs/Log';
 
 const propTypes = {
-    /** Controls whether the loader is mounted and displayed */
-    visible: PropTypes.bool,
+    /** Name used in the logs if the loader is displayed for too long time */
+    name: PropTypes.string,
+
+    /** Optional duration (ms) after which a message would be logged if the loader is still covering the screen */
+    timeout: PropTypes.number,
```

Comment:
**suggestion:** Rather than have a custom timeout here we can just make this `shouldLogAlertOnTimeout` since we use 15 seconds everywhere anyway and there are not cases where we need a more specific timeout (and maybe never will be)

Reply:
In order to capture meaningful logs we need to have `logDetail` with at least the screen name, or some name we can use to distinct log messages

Instead of introducing a flag like `shouldLogAlertOnTimeout` we use the optional `logDetail` prop - if it is provided it's an indication that we'd like to capture logs

If you prefer to have an explicit flag anyway let me know and I'll update